lib/client.dart

```diff
@@ -573,7 +572,7 @@ class Router {
   void _leave(Iterable<Route> mustLeave, Route leaveBase) {
     mustLeave.forEach((toLeave) {
       var event = new RouteLeaveEvent(toLeave);
-      toLeave._onLeaveController.add(event);
+      (toLeave as dynamic)._onLeaveController.add(event);
```

Comment:
_onLeaveController is not defined by Route but is defined by its subclasses. The alternative would be to define _onLeaveController and similar private members in Route. I'd be glad to do that if you prefer. Seems like there were three or four members with similar strong-mode errors.